addons/udes_sale_stock/tests/test_stock_picking.py

```diff
@@ -0,0 +1,66 @@
+from . import common
+
+
+class TestStockPicking(common.BaseSaleUDES):
+    @classmethod
+    def setUpClass(cls):
+        super(TestStockPicking, cls).setUpClass()
+
+        Customer = cls.env["res.partner"]
+
+        cls.customer = Customer.create(
+            {
+                "name": "Bob",
+                "street": "1 Bobs House",
+                "street2": "Bobs street",
+                "city": "The city of Bob",
+                "zip": "EN1 4LS",
+            }
+        )
+
+        cls.apple_quant = cls.create_quant(
+            cls.apple.id,
+            cls.test_location_01.id,
+            30,
+            package_id=cls.create_package().id,
+        )
+        cls.banana_quant = cls.create_quant(
+            cls.banana.id,
+            cls.test_location_02.id,
+            30,
+            package_id=cls.create_package().id,
+        )
+
+        # Create sale order
+        cls.sale = cls.create_sale(cls.customer, requested_date="2020-01-03")
+        cls.apple_sale_line = cls.create_sale_line(cls.sale, cls.apple, 15)
+        cls.banana_sale_line = cls.create_sale_line(cls.sale, cls.banana, 15)
+        cls.sale.action_confirm()
+
+        # Get the assigned pickings
+        cls.pickings = cls.sale.picking_ids.filtered(lambda x: x.state == "assigned")
+
+    def test_doesnt_warn_picking_any_previous_pickings_not_complete_sale(self):
+        """Test that no warning message is generated when all pickings
+        are at the same stage.
+        """
+        self.assertFalse(
+            self.pickings.warn_picking_any_previous_pickings_not_complete_sale(),
+            "Assert we don't get a warning when all pickings are in the same state"
+        )
+
+    def test_warn_picking_any_previous_pickings_not_complete_sale(self):
+        """Test that a warning message is generated when not all pickings
+        are at the same stage.
+        """
+        # Set the state of a picking to waiting rather than assigned
+        self.pickings[0].state = "waiting"
```

Comment:
I would suggest to test this with two pickings one available and one waiting.

Reply:
That's what we have in this situation. I'll add an assert to make it more obvious

Reply:
Added